libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -177,7 +177,7 @@ internal class MapboxTripSession(
         }
     }
 
-    private val mainJobController: JobControl = threadController.getMainScopeAndRootJob()
+    private val mainJobController: JobControl = threadController.getSDKScopeAndRootJob()
```

Comment:
I noticed that in this class we use `launch(start = CoroutineStart.UNDISPATCHED)`? The docs say the coroutine will be invoked from the current thread until the first suspension point. Is that what we expect? See `updateRawLocation`.

Reply:
yes, it was done on purpose. I wanted one specific call to be immediate, but it seemed risky to change dispatcher for the whole scope.

Reply:
But as far as I understand, `onNewRawLocation` will be invoked from the current thread then. Are we sure the current thread is the one we need?